python/pyarrow/tests/test_compute.py

```diff
@@ -2208,6 +2208,21 @@ def test_make_struct():
         pc.make_struct(field_names=['one', 'two'])
 
 
+def test_struct_fields_options():
+    a = pa.array([4, 5, 6], type=pa.int64())
+    b = pa.array(["bar", None, ""])
+    c = pa.StructArray.from_arrays([a, b], ["a", "b"])
+    arr = pa.StructArray.from_arrays([a, c], ["a", "c"])
+
+    assert pc.struct_field(arr,
+                           indices=[1, 1]) == pa.array(["bar", None, ""])
+    assert pc.struct_field(arr,
+                           indices=[0]) == pa.array([4, 5, 6], type=pa.int64())
+    assert pc.struct_field(arr, indices=[]) == arr
+
+    # errors pc.struct_field(arr)
```

Comment:
Can you maybe add a `TODO: https://issues.apache.org/jira/browse/ARROW-14853 to all default options`? (then I don't think this PR necessarily needs to wait until ARROW-14853  is fixed

Reply:
Sure, will do.

Reply:
Should I add `StructFieldOptions` in `test_option_class_equality()` also?
https://github.com/apache/arrow/blob/bec0a6beac3415e03400088a3269f3be3fc7150c/python/pyarrow/tests/test_compute.py#L119-L120